src/client.rs

```diff
@@ -240,53 +307,55 @@ impl Client {
         &mut self,
         image_ref: &Reference,
         image_data: &ImageData,
-        config_data: &[u8],
```

Comment:
This API feels cumbersome for users pushing something that they haven't just pulled, as in that case they need to create an ImageData. 

perhaps it would be better as the following? 
```
pub async fn push(
        &mut self,
        image_ref: &Reference,
        layers: Vec<ImageLayer>,
        config: Config,
        auth: &RegistryAuth,
        manifest: Option<OciImageManifest>)
```

Reply:
Yes, I think it would provide a better UX